pipeline_tools/create_envelope.py

```diff
@@ -1,277 +1,416 @@
+    return subject_url
 
 
 def get_envelope_url(submit_url, auth_headers, http_requests):
-    """Query the ingest service to get the url for creating envelopes.
+    """Query the Ingest service to get the url for creating envelopes.
 
     Args:
-        submit_url (str): the root url of the ingest service
-        auth_headers (dict): headers to use for auth
-        http_requests (HttpRequests): the HttpRequests object to use
+        submit_url (str): The root url of the Ingest service for submissions.
+        auth_headers (dict): Dict representing headers to use for auth.
+        http_requests (http_requests.HttpRequests): The HttpRequests object to use for talking to Ingest.
 
     Returns:
-        envelope_url (str): A string giving the url for creating envelopes
+        envelope_url (str): A string giving the url for creating envelopes.
 
     Raises:
-        requests.HTTPError: for 4xx errors or 5xx errors beyond timeout
+        requests.HTTPError: For 4xx errors or 5xx errors beyond timeout.
     """
     print('Getting envelope url from {}'.format(submit_url))
-    response = http_requests.get(submit_url,
-                                 headers=auth_headers)
-    envelope_url = get_subject_url(response.json(), 'submissionEnvelopes')
+    response = http_requests.get(submit_url, headers=auth_headers)
+    envelope_url = get_subject_url(endpoint_dict=response.json(), subject='submissionEnvelopes')
     return envelope_url
 
 
 def create_submission_envelope(envelope_url, auth_headers, http_requests):
-    """Creates a new submission envelope
+    """Creates a new submission envelope and return its content.
 
     Args:
-        envelope_url (str): the url for creating envelopes
-        auth_headers (dict): headers to use for auth
-        http_requests (HttpRequests): HttpRequests object to use
+        envelope_url (str): The url for creating envelopes.
+        auth_headers (dict): Dict representing headers to use for auth.
+        http_requests (http_requests.HttpRequests): The HttpRequests object to use for talking to Ingest.
 
     Returns:
-        envelope_js (dict): Dict representing the JSON response to the request
+        envelope_dict (dict): Dict representing the JSON response to the request of creating the envelope.
 
     Raises:
-        requests.HTTPError: for 4xx errors or 5xx errors beyond timeout
+        requests.HTTPError: For 4xx errors or 5xx errors beyond timeout.
     """
     print('Creating submission envelope at {0}'.format(envelope_url))
-    response = http_requests.post(envelope_url,
-                                  '{}',
-                                  headers=auth_headers)
-    envelope_js = response.json()
-    return envelope_js
+    response = http_requests.post(envelope_url, '{}', headers=auth_headers)
+    envelope_dict = response.json()
+    return envelope_dict
+
+
+# FIXME: questionable in new bundle structure :(
```

Comment:
@rexwangcc we can confirm that this is correct by checking what the new submission envelope looks like

Reply:
Sorry for the confusion, I have removed the fixmes in a newer commit, but haven't push them yet. Nice catch!